### DISCUSSION TITLE
Calling FilterFn through Virtual Keyboard

### DISCUSSION THREAD

**Discussion by jmriyaz84** [quasarframework/quasar#11808] [General - Components / Directives / etc] [answered]
I have developed the data entry form using quasar & I am using q-select with autofilter using @filter="filterFn". It is working fine as expected when I am using inbuilt keyboard. But We have developed virtual keypad , so please guide me how to call @filter="filterFn" through virtual keypress.

**Q-Select**
`<q-select
            name="searchTxt"
            v-model="item"
            :options="measurements"
            borderless
            behavior="menu"
            dense
            @filter="filterFn"
            map-options
            use-input
            emit-value
            @input="value => onSearchChange(value)"
          >
            <template v-slot:prepend>
              <q-icon name="search" />
            </template>
            <template v-slot:append>
              <q-icon
                v-if="item"
                name="close"
                @click.stop="
                  () => {
                    item = '';
                    category = 'daily';
                    tab = 'daily';
                    measurements = actualMeasurements;
                  }
                "
                class="cursor-pointer"
                size="xs"
              />
            </template>
            <template v-slot:no-option>
              <q-item>
                <q-item-section class="text-grey">
                  No Measurements Found
                </q-item-section>
              </q-item>
            </template>
          </q-select>`

**Filter Function**
`  filterFn(val, update, abort) {
      if (val.length < 1) {
        abort();
        return;
      }
      update(() => {
        this.measurements = _.filter(this.actualMeasurements, measurement => {
          return measurement.label.toUpperCase().includes(val.toUpperCase());
        });
      });
    },`

**Comment by pdanpdan** [comment]
The virtual keyboard should work exactly the same as a physical keyboard.

**Reply by jmriyaz84** [reply]
When it is text box I can assign the pressed key value to model , but I don't know how to show Q-Select auto filter while key press.

**Comment by pdanpdan** [comment]
Check `updateInputValue` in the docs

**Reply by jmriyaz84** [reply]
Thanks @pdanpdan for your quick response . Actually while any key press I am calling callBack function named as 'addKey(key)' which gets key as input.
Inside that addKey function I think I have to call filterFn(val, update, abort) & I can pass key as val but how to pass update & abort function ? Please guide.

**Reply by jmriyaz84** [reply]
@pdanpdan  I have created codepen for this  . [https://codepen.io/jmriyaz84/pen/vYedYeV?editors=1011](url)

Here I have added virtual keyboard,  when I click any button(in virtual keyboard)  the select box is not getting populated automatically. Please guide.

**Comment by pdanpdan** [comment, marked as answer]
https://codepen.io/pdanpdan/pen/RwLQWPJ?editors=1011

**Reply by jmriyaz84** [reply]
@pdanpdan  Thanks for your help , you have saved by time. Thanks a lot :)

**Reply by jmriyaz84** [reply]
@pdanpdan , I was struggling before because I was not aware about this.$refs.select.updateInputValue & this.$refs.select.showPopup functions , I can't find documentation about this in Quasar site.

Can you please tell me where I can find documentation for $refs , updateInputValue  & showPopup .

**Reply by MilosPaunovic** [reply]
On top of the page for every component  in the docs there is an `API` section, which has few tabs - `props`, `slots`, `event`, `methods`... That is the place for you to search everything available to the certain component.

[https://quasar.dev/vue-components/select#qselect-api](https://quasar.dev/vue-components/select#qselect-api)

**Reply by jmriyaz84** [reply]
Thanks Got it.
